cpp/tests/replace/replace_tests.cu

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2018-2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <tests/utilities/legacy/cudf_test_fixtures.h>
+#include <tests/utilities/legacy/cudf_test_utils.cuh>
+#include <tests/utilities/legacy/column_wrapper.cuh>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/replace.hpp>
+
+#include <cudf/cudf.h>
+
+#include <gtest/gtest.h>
+
+// This is the main test feature
+template <class T>
+struct ReplaceTest : public GdfTest
+{
+
+  ReplaceTest()
+  {    
+  }
+
+  ~ReplaceTest()
+  {
+  }
+};
+
+// ugly.  only here to get around strict-aliasing compiler warnings
+uint32_t hard_cast_float(float val)
+{
+   void *v_val = &val;
+   return *((uint32_t*)v_val);
+}
+uint64_t hard_cast_double(double val)
+{
+   void *v_val = &val;
+   return *((uint64_t*)v_val);
+}
+
+// Test for normalize_nans_and_nulls
+TEST(ReplaceTest, NormalizeNansAndZeros)
+{      
+   int num_els = 8;   
+   
+   // floats
+   {
+      float els[8] = { 32.5f, -0.0f, 111.0f, -NAN, NAN, 1.0f, 0.0f, 54.3f };   
+
+      // copy the data to mutable device column
+      auto test_data = cudf::make_numeric_column(cudf::data_type(cudf::FLOAT32), num_els, cudf::ALL_VALID, 0);
+      auto test_data_mview = test_data->mutable_view();
+      cudaMemcpy(test_data_mview.head(), els, sizeof(float) * num_els, cudaMemcpyHostToDevice);
```

Comment:
Use `cudf::column_wrapper` to avoid explicit memcpy?

Reply:
Fixed in ff7f197e70fcccca7539a90eec5fade7f986e36f